#1385 Generate constants for path in spring boot @RequestMapping

bilak opened the pull request:
<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. 

@bbdouglas  @sreeshas  @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608 @martin-mfg 

closes #1385

martin-mfg commented:
Hi @bilak, thanks for the PR!

Instead of introducing a new lambda "lambda.camelCaseToSnakeCase", would it work to use the already existing ["lambda.snakecase"](https://github.com/OpenAPITools/openapi-generator/blob/cfe6520283b6b03c28a7b1704d9d56d1e6cb67ee/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L410)?

And could you take care of the failing tests please?

bilak commented:
@martin-mfg I'm not sure if I rebased somehow wrong or why 5k files is in this commit. I reverted back my custom lambda and fixed the test...let me know if I should do something additionally

bilak commented:
I've seen also [this one](https://github.com/OpenAPITools/openapi-generator/pull/1532) I've tried that route but there are some issues in some codegen which I honestly don't udnerstand.

elagerho commented:
@bilak Any updates on getting this merged? 🙏

bilak commented:
@wing328 

I've executed
```
./mvnw clean package 
./bin/generate-samples.sh ./bin/configs/*.yaml
./bin/utils/export_docs_generators.sh
```

but tests are still failing...what should I execute in addition?
